src/status_im/models/browser-history.cljs

```diff
@@ -0,0 +1,50 @@
+(ns status-im.models.browser-history
+  (:require [re-frame.core :as re-frame]))
+
+(defn dont-store-history-on-nav-change? [db]
+  (get-in db [:browser/options :dont-store-history-on-nav-change?]))
+
+(defn dont-store-history-on-nav-change []
+  (re-frame/dispatch [:update-browser-options {:dont-store-history-on-nav-change? true}]))
+
+(defn clear-dont-store-history-on-nav-change []
+  (re-frame/dispatch [:update-browser-options {:dont-store-history-on-nav-change? false}]))
+
+(defn dont-store-history-on-nav-change-if-history-exists [db browser-id]
+  (let [browsers (get-in db [:browser/browsers])
+        browser (get browsers browser-id)]
+    (hash-map :dont-store-history-on-nav-change? (some? (:history browser)))))
+
+(defn back [browser]
+  (let [back-index (dec (:history-index browser))
+        back-url (nth (:history browser) back-index)]
+    (dont-store-history-on-nav-change)
+    (re-frame/dispatch [:update-browser (-> browser (assoc :url back-url) (assoc :history-index back-index))])))
+
+(defn forward [browser]
+  (let [forward-index (inc (:history-index browser))
+        forward-url (nth (:history browser) forward-index)]
+    (dont-store-history-on-nav-change)
+    (re-frame/dispatch [:update-browser (-> browser (assoc :url forward-url) (assoc :history-index forward-index))])))
+
+(defn can-go-back? [browser]
+  (not= (:history-index browser) 0))
+
+(defn can-go-forward? [browser]
+  (< (:history-index browser) (dec (count (:history browser)))))
+
+(defn record-history-in-browser-if-needed [db raw-browser url loading]
+  (let [browser (assoc raw-browser :url url)]
+    (cond
+      loading browser
+      (dont-store-history-on-nav-change? db)
+      (do (clear-dont-store-history-on-nav-change)
```

Comment:
there is no reason to "hide" dispatch under this function , better to dispatch right here. will be more readable

Reply:
and tbh it's not good to have not pure function in model, we need to move all side effects (event dispatching) outside this function

Reply:
@flexsurfer while I agree that is not ideal not having side effects in model, I think there are things that trumps that and are more important, for example that code needs not to be fragmented, as it makes it prone to errors and hard to follow.

If we want to avoid side-effects in model, dependency injection like patters (i.e. passing a function `on-blah`), can be used, or if we can of course ::do-stuf/:dispatch/:dispatch-n.

In this case looks like we can just operate directly on db, and just return an updated db, worst case scenario if we really need a dispatch `:dispatch` can be added.

Reply:
@flexsurfer @cammellos Since you're not unanimous on the best way forward, if it's ok with you, I'll leave this one as-is. :beers: